str/str_iterative_eh_runner.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python3
+
+"""
+This script uses ExpansionHunterv5 to call STRs on WGS cram files.
+Required input: --ehregions (file path to variant catalog) and external sample IDs 
+For example: 
+analysis-runner --access-level test --dataset tob-wgs --description "tester" --output-dir "tester" str_iterative_eh_runner.py --ehregions=gs://cpg-tob-wgs-test/hoptan-str/Illuminavariant_catalog.json TOB1XXXX TOB1XXXX
+
+Required packages: sample-metadata, hail, click, os 
+pip install sample-metadata hail click
+
+"""
+import os
+import hailtop.batch as hb
+import click
+
+from sample_metadata.model.analysis_type import AnalysisType
+from sample_metadata.model.analysis_query_model import AnalysisQueryModel
+from sample_metadata.apis import AnalysisApi, SampleApi
+from sample_metadata.models import AnalysisStatus
+from cpg_utils.config import get_config
+
+from cpg_utils.hail_batch import (
+    authenticate_cloud_credentials_in_job,
+    copy_common_env,
+    remote_tmpdir,
+    output_path
+)
+
+config = get_config()
+
+DATASET = config['workflow']['dataset']
+OUTPUT_SUFFIX = config['workflow']['output_prefix']
+BILLING_PROJECT = config['hail']['billing_project']
+REF_FASTA = os.path.join(config['workflow']['reference_prefix'], 'hg38/v0/Homo_sapiens_assembly38.fasta')
+SAMTOOLS_IMAGE = os.path.join(config['workflow']['image_registry_prefix'], 'samtools:v0')
+EH_IMAGE = os.path.join(config['workflow']['image_registry_prefix'], 'expansionhunter:5.0.0')
+
+#inputs: 
+#variant catalog
+@click.option('--ehregions', help='Full path to Illumina Variants catalog')\
+#input TOB ID 
+@click.argument('tob-wgs-ids', nargs=-1)
+@click.command()
+
+def main(ehregions, tob_wgs_ids: list[str]):  # pylint: disable=missing-function-docstring 
+   # Initializing Batch
+    backend = hb.ServiceBackend(
+        billing_project=get_config()['hail']['billing_project'],
+        remote_tmpdir=remote_tmpdir(),
+    )
+    b = hb.Batch(backend=backend, default_image=os.getenv('DRIVER_IMAGE'))
+
+    tob_wgs_to_cpg_sample_id_map: dict[str, str] = SampleApi().get_sample_id_map_by_external('tob-wgs',list(tob_wgs_ids))
+    cpg_sample_id_to_tob_wgs_id = {cpg_id: tob_wgs_id for tob_wgs_id, cpg_id in tob_wgs_to_cpg_sample_id_map.items()}
+
+    analysis_query_model = AnalysisQueryModel(
+        sample_ids=list(tob_wgs_to_cpg_sample_id_map.values()),
+        projects=['tob-wgs'],
+        type=AnalysisType("cram"),
+        status=AnalysisStatus("completed"),
+        meta={"sequence_type": "genome", "source": "nagim"}
+    )
+    crams_path = AnalysisApi().query_analyses(analysis_query_model)
```

Comment:
Suggestion here to add some code to check whether samples don't have crams:

```suggestion
    crams_path = AnalysisApi().query_analyses(analysis_query_model)
    
    cpg_sids_with_crams = set(sid for sids in crams_path["sample_id"] for sid in sids)
    cpg_sids_without_crams = (
        set(cpg_sample_id_to_tob_wgs_id.keys()) - cpg_sids_with_crams
    )
    if cpg_sids_without_crams:
        tob_wgs_sids_without_crams = ", ".join(
            cpg_sample_id_to_tob_wgs_id[sid] for sid in cpg_sids_without_crams
        )
        logger.warning(
            f"There were some samples without CRAMs: {tob_wgs_sids_without_crams}"
        )
```